marketing.js

```diff
@@ -0,0 +1,229 @@
+'use strict'
+
+const AutoDetectDecoderStream = require('autodetect-decoder-stream')
+const crypto = require('crypto')
+const csv = require('csv-parser')
+const fs = require('fs')
+const Promise = require('bluebird')
+const request = require('request-promise')
+const s3 = require('s3')
+
+fs.unlinkAsync = Promise.promisify(fs.unlink)
+
+const TIMESTAMP = /^(20[1-9][0-9])-([01][0-9])-([0-3][1-9])-([0-2][0-9])-([0-5][0-9])$/
+const EVENT = /^mktg-([a-z]+_[a-z]+)$/
+
+const AWS_ACCESS_KEY = process.env.FXA_AWS_ACCESS_KEY
+const AWS_SECRET_KEY = process.env.FXA_AWS_SECRET_KEY
+const AWS_S3_BUCKET = '*** TODO ***'
+const AWS_S3_PREFIX = '*** TODO ***'
+
+const MAX_EVENTS_PER_BATCH = 10
+const HMAC_KEY = process.env.FXA_AMPLITUDE_HMAC_KEY
+const API_KEY = process.env.FXA_AMPLITUDE_API_KEY
+
+if (process.argv.length !== 3) {
+  console.error(`Usage: ${process.argv[1]} {YYYY-MM-DD-hh-mm | LOCAL PATH}`)
```

Comment:
Not sure what frequency the data will be exported to S3 with yet, but I've optimistically written this to cope with per-minute cadence.